### DISCUSSION TITLE
Not existing operator with JSON type column

### DISCUSSION THREAD

**Discussion by LadIQe** [mikro-orm/mikro-orm#1644] [Q&A] [answered]
Hello, Idk if its bug, or I have something bad implemented, because exactly same bug was reported and fixed.

But here is the error
ERROR:  operator does not exist: text >= integer at character 68
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
STATEMENT:  select "e0".* from "price" as "e0" where "recommended"->>'withVat' >= 0 and "recommended"->>'withVat' <= 30

Im using nestjs with graphql, so my definition looks like this

Entity:
```
@Field(() => PriceItem, { nullable: true })
@Property({ type: 'json', nullable: true })
recommended?: PriceItem
```

Type:
```
@ObjectType('PriceType')
@InputType('PriceInput')
export class PriceItem {
  @Field(() => Int)
  vat: number

  @Field(() => Int)
  withVat: number

  @Field(() => Int)
  withoutVat: number
}
```

Now code:
```
await this.priceRepository.find({
  recommended: { withVat: { $gte: priceRange.min, $lte: priceRange.max } }
})
```

Tried different code with `$and` array, but same result

Any help?

Thank you

**Comment by B4nan** [comment, marked as answer]
You need to use embeddables here to let the ORM know the type of your json properties, otherwise it cannot type cast for you (and postgres won't allow you to compare strings with numbers - for postgres anything inside the JSON is a string/json unless you cast it). 

https://mikro-orm.io/docs/embeddables/#storing-embeddables-as-objects

**Reply by LadIQe** [reply]
thank you for fast reply

So I changed it like this:

Type:
```
@Embeddable()
@ObjectType('PriceItemType')
@InputType('PriceInput')
export class PriceItem {
  @Field(() => Int)
  @Property()
  vat: number

  @Field(() => Int)
  @Property()
  withVat: number

  @Field(() => Int)
  @Property()
  withoutVat: number
}
```

Entity:
```
@Field(() => PriceItem, { nullable: true })
@Embedded(() => PriceItem, { object: true, nullable: true })
recommended?: PriceItem
```

Code is same. Now I got this error:
ERROR:  syntax error at or near "(" at character 47
STATEMENT:  select "e0".* from "price" as "e0" where "e0".("recommended"->>'withVat')::float8 >= 0 and "e0".("recommended"->>'withVat')::float8 <= 30

**Reply by B4nan** [reply]
Use latest dev version, this should be already fixed in master.

**Reply by LadIQe** [reply]
Im using 4.5.1

**Reply by LadIQe** [reply]
oh yeah ok I can see that, thank you
